### ISSUE TITLE
Authpolicies execution count issues

### ISSUE THREAD

**Issue by BoSen29** [ironmansoftware/universal-dashboard#1406]
#### Describe the Issue
See the forumthread for context:
https://forums.universaldashboard.io/t/new-oauth-2-0-authentication-return-url-mismatch/1833/30

#### To Reproduce
Have auth policies -> Lot of controls -> problems

#### Expected behavior
Auth policies executed once on sigin, and cached for future reference.

#### Script Sample
See forum thread.

#### Version Information
UD 2.8 / 2.8.1 
oAuth v2

**Comment by adamdriscoll** [comment]
Can you try this on the nightly build tonight? This should now be resolved. The auth policies are being executed on the auth events.

**Comment by BoSen29** [comment]
Hi @adamdriscoll 
Yeah, the new nightly.... not good :O 
500 errors on fetch service no matter which auth method is being used.
Want me to do any more debugging? 
`
10:17:04 [Warn] Microsoft.AspNetCore.Session.SessionMiddleware Error unprotecting the session cookie.
10:17:04 [Warn] Microsoft.AspNetCore.Session.SessionMiddleware Error unprotecting the session cookie.
10:17:04 [Warn] Microsoft.AspNetCore.Session.SessionMiddleware Error unprotecting the session cookie.
10:17:04 [Error] Microsoft.AspNetCore.Server.Kestrel Connection id "0HLSFCHQI3979", Request id "0HLSFCHQI3979:00000008": An unhandled exception was thrown by the application.
10:17:04 [Error] Microsoft.AspNetCore.Server.Kestrel Connection id "0HLSFCHQI3977", Request id "0HLSFCHQI3977:00000008": An unhandled exception was thrown by the application.
10:17:04 [Error] Microsoft.AspNetCore.Server.Kestrel Connection id "0HLSFCHQI397B", Request id "0HLSFCHQI397B:00000006": An unhandled exception was thrown by the application.
10:17:04 [Warn] Microsoft.AspNetCore.Session.SessionMiddleware Error unprotecting the session cookie.`

**Comment by BoSen29** [comment]
This is pre login by the way, no login page being loaded.

**Comment by adamdriscoll** [comment]
Just spent some time looking into this. I have it working on my end but waiting for integration tests to run and I'll test the nightly once the build is done. Hopefully, this does it but I'll do some more testing before letting you know it's good to go.

**Comment by BoSen29** [comment]
Hit me up if you want me to test anything! :-) 
Happy bugfixing bro

**Comment by adamdriscoll** [comment]
I was working in a feature branch without realizing it. None of the code I was changing was actually in master. 😆 I just validated my changes and merged to master. Tonight's nightly will include them.

**Comment by BoSen29** [comment]
Glad to hear that we mortals arent the only ones doing weird user related errors :-) 
I'll test it and validate that it works ! Thanks !

**Comment by BoSen29** [comment]
Sorry bro: 
![image](https://user-images.githubusercontent.com/54361737/71878205-5e14d600-312b-11ea-8824-202f20be6b44.png)

**Comment by adamdriscoll** [comment]
Weird. Does it work and you see that error or does it not work and that's why you checked? I'll try and reproduce. I didn't look in the JS console so maybe it's just there...

**Comment by BoSen29** [comment]
Visually identical as previous, just a gray background.
Note: this is not using the oauth v2.0, but the original implementation of Azure AD Auth.
Haven't tested it on oAuth yet.

There was an “unexpected <” aswell, regarding the “<doctype> tag at the top.

I can enable some udlogging av send you if you’d like?

**Comment by adamdriscoll** [comment]
Ohhhhhh. Let me try the original Azure AD Auth. That's probably what I'm missing. 

In terms of the other error, I see that one. I will get that resolved. It was resolved in the community fork but not the enterprise. Just needs to be merged in.

**Comment by BoSen29** [comment]
Still experiencing issues with the oAuth 2 tho... 
![image](https://user-images.githubusercontent.com/54361737/71898624-13f71900-315a-11ea-9b73-57785a3e062e.png)

**Comment by BoSen29** [comment]
Now, setting the authpolicies to "return $true" won't work either on oAuth v2.0 :-/
Same errors as previously on the oAuth.

**Comment by adamdriscoll** [comment]
There is something I'm missing. Just validated and I have this working on both OAuth 2.0 and OAuth 1.0 with auth policies. Can we do a Zoom call or something? I would like to see what I'm missing and I can show you what I'm doing so we can compare notes and get to the bottom of this. 

I won't have any time this morning (I'm UTC-7) but maybe we could schedule something that works in the next couple days.

**Comment by BoSen29** [comment]
Sounds like a plan dude, 
DM me and we'll schedule something. 

Appreciate your help :-)

**Comment by adamdriscoll** [comment]
@BoSen29 - Tonight's build will fix the issue where if you return values from the endpoint that aren't true\false, it will cause the dashboard to fail to load. Give er a go tomorrow and let me know! 

I also added a new log message to the point out that the auth policy isn't returning true\false. 

I still need to fix the websocket issue.

**Comment by BoSen29** [comment]
@adamdriscoll 
Damn you're fast dude!
Soe... i tried to upgrade the dev dashboard (which we got working yesterday) and its seems like dynamic pages prop is not passed properly
![image](https://user-images.githubusercontent.com/54361737/72047600-36547800-32bb-11ea-92cb-2f4a7a20a02f.png)
Didn't have a chance to look at the new fancy log message :-/ 

When starting a unlicensed dashboard it'll just throw a no-remote error, do you want me to make a separate bug report on this?

**Comment by adamdriscoll** [comment]
Ah crap. Fixed that last night but didn't realize the build released before I fixed. Publishing a new release right now. 

Can you make a separate issue for the no-remote error?

**Comment by BoSen29** [comment]
Awesome man! 
I'll prolly give this a go later tonight and update you! 

I'll throw together some evidence and post another bug report :-)

**Comment by BoSen29** [comment]
I just created a new bug for the double websocket aswell. 
#1440
For documentation and ease of referencing.

**Comment by adamdriscoll** [comment]
Sounds like this is resolved. Wooo!
